pkg/apis/openebs.io/v1alpha1/types.go

```diff
@@ -230,3 +230,43 @@ type CStorVolumeReplicaList struct {
 
 	Items []CStorVolumeReplica `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=cstorvolume
+
+// CStorVolume describes a CStorVolume.
+type CStorVolume struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   CStorVolumeSpec   `json:"spec"`
+	Status CStorVolumeStatus `json:"status"`
+}
+
+// CStorVolumeSpec is the spec for a CStorVolume resource
+type CStorVolumeSpec struct {
+	CStorControllerIP string `json:"cstorControllerIP"`
+	VolumeName        string `json:"volumeName"`
+	VolumeID          string `json:"volumeID"`
+	Capacity          string `json:"capacity"`
+	Status            string `json:"status"`
```

Comment:
What is this Status property inside the CStorVolumeSpec vs. the one outside?

Reply:
the Status which is outside is not being used anywhere. Again this was a duplication from cstor pool watcher code. I am removing this field from the struct.